### PR TITLE
REF: share comparison methods for DTA/TDA/PA

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -37,19 +38,94 @@
...
-from pandas.core.ops.invalid import make_invalid_op
+from pandas.core.ops.invalid import invalid_comparison, make_invalid_op
 
 from pandas.tseries import frequencies
 from pandas.tseries.offsets import DateOffset, Tick
 
 from .base import ExtensionArray, ExtensionOpsMixin
 
 
+def _datetimelike_array_cmp(cls, op):
```
Comment (WillAyd) [inline pandas/core/arrays/datetimelike.py]: This isn't an exact copy of dt_array_cmp right? If not can you help point out where integrates from period / td methods?

**Review by WillAyd — COMMENTED** [review]
looks good at a glance

File: pandas/core/arrays/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py]: i put some comments in when moving this, are there specific parts that arent clear?

**Comment by jreback** [comment]
thanks @jbrockmendel 

@WillAyd this was one of the goals of the last series of PRs that @jbrockmendel did, to make identical code in the subclasses. (and now this consolidates it).

**Comment by WillAyd** [comment]
Yep makes sense

**Pull request merged**